recovery.go

```diff
@@ -0,0 +1,83 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"runtime/debug"
+)
+
+type recoveryHandler struct {
+	handler    http.Handler
+	logger     *log.Logger
+	printStack bool
+}
+
+// Option provides a functional approach to define
+// configuration for a handler; such as setting the logging
+// whether or not to print strack traces on panic.
+type Option func(http.Handler)
+
+func parseOptions(h http.Handler, opts ...Option) http.Handler {
+	for _, option := range opts {
+		option(h)
+	}
+
+	return h
+}
+
+// RecoveryHandler is HTTP middleware that recovers from a panic,
+// logs the panic, writes http.StatusInternalServerError, and
+// continues to the next handler.
+//
+// Example:
+//
+//  r := mux.NewRouter()
+//  r.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+//  	panic("Unexpected error!")
+//  })
+//
+//  recoverRouter := handlers.RecoveryHandler(r)
+//  http.ListenAndServe(":1123", recoverRouter)
```

Comment:
You could just show this on one line to simplify it - e.g. `http.ListenAndServe(":1123", handlers.RecoveryHandler(r))`